Add an opt-in option to unmarshal

jonathanstowe opened the pull request:
In some applications it might not be ideal to initialise all the
attributes that are present in the JSON (either for security or other
application reasons,) so this adds an `:opt-in` adverb to `unmarshal`
such that, if applied, only those attributes that have been explicitly
allowed (by appliction of a trait,) will be initialised.

vrurg committed:
Expose unmarshalling from deserialized data

Make it possible to `unmarshal({foo => 42}, Foo)`.

Resolves tadzik/JSON-Unmarshal#32

vrurg committed:
Try to fix travis

Use `zef` and `App::Prove6`

vrurg committed:
Preserve parameterization of arrays and hashes

Make sure that if unmarshalling into `Array[Type]` and `Hash[Type]` then
the returned object will have the same type.

jonathanstowe committed:
Add an opt-in option to unmarshal

In some applications it might not be ideal to initialise all the
attributes that are present in the JSON (either for security or other
application reasons,) so this adds an `:opt-in` adverb to `unmarshal`
such that, if applied, only those attributes that have been explicitly
allowed (by appliction of a trait,) will be initialised.

jonathanstowe commented:
There is symmetrical change in the JSON::Marshal (only serialise the marked attributes.)

vrurg committed:
Start migration to Raku Community Modules

- Update extensions to conform modern requirements
- Replace references to Perl 6 with Raku
- Update META6.json

Technical changes:

- Support for nominalizable types (definites and coercions)
- Support for de-serialized data
- Throw specialized exception instead of just `die` and provide more
  detailed error information

vrurg committed:
Remove travis config

vrurg committed:
Hand over under App::Mi6 govern

Prepare for the next release

vrurg committed:
Update Changes

vrurg committed:
0.09

vrurg commented:
@jonathanstowe I have finished migrating the module to the raku-community-modules. Would you, please, re-publish this PR over there?

vrurg committed:
Fixed another problem with nominalizables

When the nominal of a nominalizable is a non-supported class (none of
Int, Str, etc. – like, say DateTime) we were failing to obtain
attributes.

vrurg committed:
Remove JSON::Class used for debugging purposes

vrurg committed:
0.10

jonathanstowe commented:
Replaced by https://github.com/raku-community-modules/JSON-Unmarshal/pull/1
